shared/qt/idian/include/Idian/Utils.hpp

```diff
@@ -136,6 +127,36 @@ class Utils {
 			removeClass(widget, classname);
 		}
 	}
+
+	static void applyColorToIcon(QWidget *widget)
+	{
+		QAbstractButton *button = qobject_cast<QAbstractButton *>(widget);
+		if (button && !button->icon().isNull()) {
+			// Filter is on a widget with an icon set, update it's colors
+			QStyleOptionButton opt;
+			opt.initFrom(button);
+
+			QColor color = opt.palette.color(QPalette::ButtonText);
+			QPixmap tinted = recolorPixmap(button->icon().pixmap(button->iconSize(), QIcon::Normal), color);
+			QIcon tintedIcon;
+			tintedIcon.addPixmap(tinted, QIcon::Normal);
+			tintedIcon.addPixmap(tinted, QIcon::Disabled);
+
+			button->setIcon(tintedIcon);
+		}
+	}
+
+	static QPixmap recolorPixmap(const QPixmap &src, const QColor &color)
+	{
+		QImage img = src.toImage();
+		QPainter p(&img);
+		p.setCompositionMode(QPainter::CompositionMode_SourceIn);
+		p.fillRect(img.rect(), color);
+		p.end();
+		return QPixmap::fromImage(img);
+	}
+
+	void applyStateStylingEventFilter(QWidget *widget);
```

Comment:
I agree with this function not being inline, what do you think about also doing that for `applyColorToIcon` and `recolorPixmap` (and possibly some existing methods)? The header is getting quite filled with inline function that don't exactly have to be inline.